File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    const answer = Array(prices.length).fill(0);
    const info = [];

    for (let i = 0; i < prices.length; i++) {
        for (const [price, time] of info) {
            answer[time]++;
        }

        while (info.length && prices[i] < info[info.length - 1][0]) {
            info.pop();
        }

        info.push([prices[i], i]);
    }

    return answer;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0 for _ in range(len(prices))]
    check = []

    for i, p in enumerate(prices):
        while check and prices[check[-1]] > p:
            j = check.pop()
            answer[j] = i - j

        check.append(i)

    while check:
        j = check.pop(0)
        answer[j] = len(prices) - j - 1

    return answer